fetch.bs

```diff
@@ -6710,6 +6748,362 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+    <a for=navigable>parent</a>'s <a>active document</a> and <var>document</var>
+    <a>share deferred-fetch quota</a>, then decrement <var>quota</var> by
+    <var>otherNavigable</var>'s <a>navigable container</a>'s
+    <a>subframe reserved deferred-fetch quota</a>.
+
+    <p class=note>This is a cross-origin/cross-agent subframe, so it spends some of the quota of
+    this document and its relatives.
+  </ol>
+ <li><p>If <var>quota</var> is less than zero, than return zero.
+
+ <li><p>If <var>quota</var> is less than <var>quotaForOrigin</var>, then
+ return <var>quota</var>.
+
+ <li><p>Return <var>quotaForOrigin</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To <dfn export>reserve deferred-fetch quota</dfn> for a <a>navigable container</a>
+<var>container</var> given an <a for=/>origin</a> <var>originToNavigateTo</var>:
+
+<p class=note>This is called when the container document initiates a navigation. It potentially
+reserves either 64kb or 8kb of quota for the frame, if it doesn't <a>share deferred-fetch quota</a>
+with the container and the permissions policy allow. It is not observable to the cotnainer document
+whether the reserved quota was used in practice.
+This algorithm assumes that the container's document might delegate quota to the navigated frame,
+and the reserved quota would only apply in that case, and would be ignored if it ends up being
+shared.
+
+<ol>
+ <li><p>If the <a data-lt="define an inherited policy for feature in container">inherited policy</a>
+ for {{PermissionPolicy/"deferred-fetch"}}, <var>container</var> and <var>originToNavigateTo</var>
+ is <code>Enabled</code>, and the <a>available deferred-fetch quota</a> for
+ <var>container</var>'s <a>container document</a> is equal or greater than
+ <a>optional subframe deferred-fetch quota</a>, then set <var>container</var>'s
+ <a>reserved deferred-fetch quota</a> to <a>optional subframe deferred-fetch quota</a>.
+
+ <li>
+  <p>Otherwise, if the
+  <a data-lt="define an inherited policy for feature in container">inherited policy</a> for
+  {{PermissionPolicy/"deferred-fetch-minimal"}}, <var>container</var> and
+  <var>originToNavigateTo</var> is <code>Enabled</code>, then:
+  <ol>
+   <li><p>Let <var>delegatedQuota</var> be <a>deferred-fetch delegated quota</a>.
+
+   <li>
+    <p><a for=list>For each</a> <var>navigable</var> of <var>container</var>'s
+    <a>node navigable</a>'s <a for=navigable>top-level traversable</a>'s
+    <a>inclusive descendant navigables</a> who meets all of the following conditions:
+    <ul class=brief>
+     <li><p><var>navigable</var> is not <var>container</var>'s <a>content navigable</a>.
+
+     <li><p><var>navigable</var>'s <a>active document</a> and <var>navigable</var>'s
+     <a for=navigable>top-level traversable</a>'s <a>active document</a> do not
+     <a>share deferred-fetch quota</a>.
+
+     <li><p><var>navigable</var>'s <a>navigable container</a>'s <a>node document</a> and
```

Comment:
```
navigable’s navigable container’s node document and navigable’s top-level traversable’s active document share deferred-fetch quota.
```

What is `navigable’s navigable container’s node document`, given `navigable` = `iframe` below?

```
<iframe src="iframe-1.html"></iframe>
```

Reply:
The iframe's parent document

Reply:
It's a little unclear to me how it is translated to parent document.. is the following correct?

* `navigable` = iframe's document
* `navigable's navgable container` = iframe
* `navigable’s navigable container’s node document` = ???

Reply:
`navigable` = the iframe window (or `Frame` in blink)
`navigable's navigable container` = the iframe element, `FrameOwner` in blink) 
`navigable’s navigable container’s node document` = the iframe element's document, so it's the document where the `<iframe>` element is, not its internal `contentDocument`.

Reply:
Ah thanks. I was confused between `node navigable` and `node document`